apps/passport-client/components/modals/AccountImportModal.tsx

```diff
@@ -0,0 +1,187 @@
+import { Spacer } from "@pcd/passport-ui";
+import { PCD } from "@pcd/pcd-types";
+import { useCallback, useEffect, useState } from "react";
+import styled from "styled-components";
+import { useFilePicker } from "use-file-picker";
+import { useDispatch, usePCDCollection } from "../../src/appHooks";
+import { Button } from "../core";
+
+type ImportState =
+  | {
+      valid: true;
+      imported: false;
+      pcds: PCD[];
+      folders: FolderMap;
+      conflicts: number;
+    }
+  | { valid: true; imported: true; added: number }
+  | { valid: false };
+
+type FolderMap = { [pcdId: string]: string };
+
+export function AccountImportModal() {
+  const { openFilePicker, filesContent } = useFilePicker({
+    accept: ".json",
+    multiple: false
+  });
+
+  const pcdCollection = usePCDCollection();
+  const dispatch = useDispatch();
+
+  const [importState, setImportState] = useState<ImportState | undefined>();
+
+  const importPCDs = useCallback(() => {
+    // Should never happen, but makes TypeScript happy that we checked for it
+    if (!importState.valid || importState.imported === true) return;
+
+    const addedPCDs: PCD[] = [];
+    for (const pcd of importState.pcds) {
+      if (!pcdCollection.hasPCDWithId(pcd.id)) {
+        try {
+          pcdCollection.add(pcd);
+          if (importState.folders[pcd.id]) {
+            pcdCollection.setFolder(pcd.id, importState.folders[pcd.id]);
+          }
+
+          addedPCDs.push(pcd);
+        } catch (e) {
+          console.log(e);
+        }
+      }
+
+      setImportState({ valid: true, imported: true, added: addedPCDs.length });
+    }
+  }, [importState, pcdCollection]);
+
+  useEffect(() => {
+    (async () => {
+      if (filesContent.length > 0) {
+        try {
+          const file = filesContent[0];
+          const data = JSON.parse(file.content);
+
+          if (data.pcds) {
+            const serializedPcdCollection = JSON.parse(data.pcds);
+            if (
+              serializedPcdCollection.pcds &&
+              serializedPcdCollection.folders
+            ) {
+              const pcds = await pcdCollection.deserializeAll(
+                serializedPcdCollection.pcds
+              );
+
+              const folders: { [pcdId: string]: string } =
```

Comment:
I'd highly recommend splitting out the functional logic of import from the UI, and writing some unit tests for it.

Reply:
This is now done - `PCDCollection` now has a `merge()` method, with tests. If we want to do the same for subscriptions then I'll add a similar method to `SubscriptionManager`.

Reply:
I'll want that for sync, but that can be a separate decision from whether we want it for import.